src/pj_internal.c

```diff
@@ -134,36 +136,193 @@ size_t pj_strlcpy(char *dst, const char *src, size_t siz) {
+            c[j] = ' ';
+
+        if (isspace (c[j]) || ';'==c[j] || '\\'==c[j]) {
+            if (0==ws && (i > 0))
+                c[i++] = ' ';
+            ws = 1;
+            continue;
+        }
+        else
+            ws = 0;
+        c[i++] = c[j];
+    }
+    c[i] = 0;
+    n = strlen(c);
+
+    /* Then make ',' and '=' greedy */
+    for (i = j = 0; j < n; j++) {
+        if (i==0) {
+            c[i++] = c[j];
+            continue;
+        }
+
+        /* Skip space before '='/',' */
+        if ('='==c[j] || ','==c[j]) {
+            if (c[i - 1]==' ')
+               c[i - 1] = c[j];
+            else
+                c[i++] = c[j];
+            continue;
+        }
+
+        if (' '==c[j] && ('='==c[i - 1] || ','==c[i - 1]) )
+            continue;
+
+        c[i++] = c[j];
+    }
+    c[i] = 0;
+    return c;
 }
 
 
 
-/* stuff below is *not* considered API, and will be moved to an "internal plumbing toolset" */
+
+/***************************************************************************************/
+size_t pj_trim_args (char *args) {
+/****************************************************************************************
+    Trim an argument string and count its number of elements.
+****************************************************************************************/
+    size_t i, m, n;
+    pj_shrink (args);
+    n = strlen (args);
+    if (n==0)
+        return 0;
+    for (i = m = 0;  i < n;  i++) {
+        if (' '==args[i]) {
+            args[i] = 0;
+            m++;
+        }
+    }
+    return m + 1;
+}
+
+
+/***************************************************************************************/
+char **pj_trimmed_args_to_argv (size_t argc, char *args) {
+/****************************************************************************************
+    Create an argv-style array from elements placed in args.
+
+    args is a trimmed string as returned by pj_trim_args(), and argv is the number of
+    trimmed strings found (i.e. the return value of pj_trim_args()).
+****************************************************************************************/
+    size_t i, j;
+    char **argv;
+    /* turn the massaged input into an array of strings */
+    argv = (char **) calloc (argc, sizeof (char *));
+    if (0==argv)
+        return 0;
+    argv[0] = args;
+    for (i = 0, j = 1;  ;  i++) {
+        if (0==args[i]) {
+            argv[j++] = args + (i + 1);
+        }
+        if (j==argc)
+            break;
+    }
+    return argv;
+}
+
+
+
+/***************************************************************************************/
+char *pj_free_format_from_argc_argv (size_t argc, const char **argv) {
+/****************************************************************************************
+    Convert free format command line input to something proj_create can eat.
+****************************************************************************************/
+    size_t i, n;
+    char *p;
```

Comment:
See `CONTRIBUTING.md`, section "Programming language":

> PROJ.4 is developed strictly in ANSI C 89.

Which means variables can't be declared within loop headers.

Compilers are also able to detect if variables are used uninitialized, so I think this can go either way.

Reply:
@schwehr I agree that uninitialized variables are in general horrible. Zeroing them at definition time (in cases where a proper value can first be computed later) is, however, not always the proper thing to do, since the compiler may warn about unused assignments, and we treat warnings as errors in all builds.